### ISSUE TITLE
Not showing any order history

### ISSUE THREAD

**Issue by Blacktempel** [philipmulcahy/azad#128]
Hi,
it seems like I have the same issue as #127.

The readme stuff did not help.
Here is the log:

    inject.ts:86 getYears() returning  Promise {<fulfilled>: Array(11)}__proto__: Promisecatch: ƒ catch()constructor: ƒ Promise()finally: ƒ finally()then: ƒ then()Symbol(Symbol.toStringTag): "Promise"__proto__: Object[[PromiseState]]: "fulfilled"[[PromiseResult]]: Array(11)0: 20201: 20192: 20183: 20174: 20165: 20156: 20147: 20138: 20129: 201110: 2010length: 11__proto__: Array(0)
    order.ts:707 checkComplete() actual:0 expected:null
    request_scheduler.ts:176 Queuing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=0&language=en_GB with 0
    request_scheduler.ts:222 Executing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=0&language=en_GB with queue size 0 and priority 00000
    order.ts:707 checkComplete() actual:0 expected:null
    order.ts:707 checkComplete() actual:0 expected:null
    request_scheduler.ts:331 Finished https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=0&language=en_GB with queue size 0
    order.ts:643 Found 69 orders for 2020
    order.ts:675 sending request for order: 0 onwards
    request_scheduler.ts:176 Queuing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=0&language=en_GB with 0
    request_scheduler.ts:222 Executing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=0&language=en_GB with queue size 0 and priority 2
    order.ts:675 sending request for order: 10 onwards
    request_scheduler.ts:176 Queuing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=10&language=en_GB with 0
    request_scheduler.ts:222 Executing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=10&language=en_GB with queue size 0 and priority 2
    order.ts:675 sending request for order: 20 onwards
    request_scheduler.ts:176 Queuing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=20&language=en_GB with 0
    request_scheduler.ts:222 Executing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=20&language=en_GB with queue size 0 and priority 2
    order.ts:675 sending request for order: 30 onwards
    request_scheduler.ts:176 Queuing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=30&language=en_GB with 0
    request_scheduler.ts:222 Executing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=30&language=en_GB with queue size 0 and priority 2
    order.ts:675 sending request for order: 40 onwards
    request_scheduler.ts:176 Queuing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=40&language=en_GB with 0
    request_scheduler.ts:222 Executing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=40&language=en_GB with queue size 0 and priority 2
    order.ts:675 sending request for order: 50 onwards
    request_scheduler.ts:176 Queuing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=50&language=en_GB with 0
    order.ts:675 sending request for order: 60 onwards
    request_scheduler.ts:176 Queuing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=60&language=en_GB with 1
    request_scheduler.ts:222 Executing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=50&language=en_GB with queue size 1 and priority 2
    request_scheduler.ts:222 Executing https://www.amazon.de/gp/css/order-history?opt=ab&digitalOrders=1&unifiedOrders=1&returnTo=&orderFilter=year-2020&startIndex=60&language=en_GB with queue size 0 and priority 2
    order.ts:707 checkComplete() actual:0 expected:69
    order.ts:714 resolving order_promises for 2020
    order.ts:716 resolved order_promises for 2020
    table.ts:498 amazon_order_history_table.displayOrders starting
    table.ts:500 amazon_order_history_table.displayOrders then func starting
    table.ts:382 amazon_order_history_table.reallyDisplayOrders starting
    table.ts:318 adding table
    table.ts:320 added table
    table.ts:471 azad.reallyDisplayOrders returning
    table.ts:504 amazon_order_history_table.displayOrders then func returning table promise.

Any idea ?

**Comment by philipmulcahy** [comment]
this _looks_ like you're partially logged out, but the code is supposed to detect that and prompt you to log back in.

Would you mind trying logging out and back in to amazon one more time?

Next: try this out:
i) disable the extension in chrome://settings
ii) enabling developer mode
iii) download this: www.mulcahyfamily.org/azad/azad.afaaa52d.zip
iv) unpack it
v) point chrome://extensions at the resulting folder (load unpacked extension)
vi) try it out. If it works, please let me know.
vii) Don't forget to disable the experimental one and enable the "webstore" version when you're done!

**Comment by alumcreek** [comment]
First off thank you so much for making this extension! With Amazon having disabled the Amazon Order History option on the accounts page this really helps a lot!
I did run into a problem related to the one in this thread where my query returned an empty table. 
I was able to successfully run the extension on my regular Amazon accounts, and it worked as expected. But when I switched accounts to my business account it kept returning empty tables. I followed your directions on what to do when this happens (many times! :-) and I had the same result, empty tables every time. I was using this link: https://www.amazon.com/gp/css/order-history
I tried your solution above with developer mode extension. I followed those directions and it also did not work. 

It looks to me like Amazon is logging me out while trying to run the report on this account - I tested it by switching back to my original account and it working fine there. Unfortunately I could not get the extension to produce a log file to share (I am on a Mac 10.14.6 Mojave and Chrome Version 86.0.4240.75.

I did find a solution though!

I have the Chrome based browser Vivaldi also installed. I installed your extension on Vivaldi (the process is the same there as on Chrome) and it actually worked on the business account. Vivaldi is only Mac based. But if someone is reading here and they have an alternative Chrome based browser on their device they might try running this extension in that other browser. 

So thank you!
